core/lib/cart/index.ts

```diff
@@ -0,0 +1,64 @@
+'use server';
+
+import { unstable_expireTag } from 'next/cache';
+import { cookies } from 'next/headers';
+
+import { addCartLineItem, AddCartLineItemsInput } from '~/client/mutations/add-cart-line-item';
+import { createCart, CreateCartInput } from '~/client/mutations/create-cart';
+import { getCart } from '~/client/queries/get-cart';
+import { TAGS } from '~/client/tags';
+
+import { MissingCartError } from './error';
+
+export async function getCartId(): Promise<string | undefined> {
+  const cookieStore = await cookies();
+  const cartId = cookieStore.get('cartId')?.value;
+
+  return cartId;
+}
+
+export async function setCartId(cartId: string): Promise<void> {
+  const cookieStore = await cookies();
+
+  cookieStore.set('cartId', cartId, {
+    httpOnly: true,
+    secure: process.env.NODE_ENV === 'production',
+    sameSite: 'lax',
+    path: '/',
+  });
+}
+
+export async function clearCartId(): Promise<void> {
+  const cookieStore = await cookies();
+
+  cookieStore.delete('cartId');
+}
+
+export const addToOrCreateCart = async (
+  data: CreateCartInput | AddCartLineItemsInput['data'],
+): Promise<void> => {
```

Comment:
```suggestion
export async function addToOrCreateCart(
  data: CreateCartInput | AddCartLineItemsInput['data'],
): Promise<void> {
```

Can we use an `async function` here instead of a const to match the style of the rest of the file?